### PR TITLE
Fix checkImage icon jumping cells on swipes

### PR DESCRIPTION
**Pull request by lucyxubroad**
Fixed bug where the checkImageView was jumping cells on swiping through polls in the User view of old poll sessions. 
- Added check for correct answer outside of MCResultCell configuration and set checkImageView's image to nil if not the correct answer
- Reseted showCorrectAnswer boolean on reuse of cells
- Moved layout of checkImageView into separate function so that it can be used both in the case where didLayoutConstraints is true and false
- Use width constraints for all optionalLabels because Swift was getting confused between setting the trailingAnchor for non correct answers and width for correct answers

### CODE REVIEW

File: Clicker/Views/Cards/MCResultCell.swift [hunk 1]
```diff
@@ -92,9 +92,23 @@ class MCResultCell: UICollectionViewCell {
     }
     
     override func updateConstraints() {
+
```
Comment (mindylou) [inline Clicker/Views/Cards/MCResultCell.swift]: remove empty line

File: Clicker/Views/Cards/MCResultCell.swift [hunk 2]
```diff
@@ -145,6 +151,16 @@ class MCResultCell: UICollectionViewCell {
         super.updateConstraints()
     }
     
+    func updateCheckImageView() {
+
```
Comment (mindylou) [inline Clicker/Views/Cards/MCResultCell.swift]: remove empty line

File: Clicker/Views/Cards/MCResultCell.swift [hunk 3]
```diff
@@ -189,6 +206,11 @@ class MCResultCell: UICollectionViewCell {
         }
     }
     
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        self.showCorrectAnswer = false
```
Comment (mindylou) [inline Clicker/Views/Cards/MCResultCell.swift]:
```suggestion
        showCorrectAnswer = false
```

File: Clicker/Views/Cards/MCResultCell.swift [hunk 4]
```diff
@@ -92,9 +92,23 @@ class MCResultCell: UICollectionViewCell {
...
+        guard let optionLabelText = optionLabel.text else { return }
+        let optionLabelWidth = optionLabelText.width(withConstrainedHeight: bounds.height, font: optionLabel.font)
+        let maxWidth = bounds.width - numSelectedLabelWidth - optionLabelHorizontalPadding * 4 - checkImageViewLength
+        
         // If we already layed out constraints before, we should only update the
         // highlightView width constraint
         if didLayoutConstraints {
+            
+            optionLabel.snp.updateConstraints { make in
+                make.width.equalTo((optionLabelWidth >= maxWidth || !showCorrectAnswer) ? maxWidth : optionLabelWidth)
```
Comment (kevinchan159) [inline Clicker/Views/Cards/MCResultCell.swift]: i would change it to:
```
let someBool = optionLabelWidth >= maxWidth || !showCorrectAnswer
make.width.equalTo(someBool ? maxWidth : optionLabelWidth)
```
so thats its more readable.

File: Clicker/Views/Cards/MCResultCell.swift [hunk 5]
```diff
@@ -160,7 +172,7 @@ class MCResultCell: UICollectionViewCell {
             if let correctAnswer = correctAnswer {
                 if correctAnswer != "" {
                     if answer == correctAnswer {
-                        showCorrectAnswer = true
+                        self.showCorrectAnswer = true
```
Comment (mindylou) [inline Clicker/Views/Cards/MCResultCell.swift]: remove `self`

**Comment by kevinchan159** [comment]
also @lucyxubroad make sure to squash commits before you merge

**Pull request merged**